tests/sphere_test.cpp

```diff
@@ -0,0 +1,120 @@
+#include "obj/objects.h"
+#include "gtest/gtest.h"
+
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+using objects::Sphere;
+
+const float kSkipFreq = 1000;
+const Vec3 kVecZero(0, 0, 0);
+
+float deg2rad(float deg) { return 2 * constants::kPi * deg / 360; }
+
+TEST(SphereCollisionTest, RayHitFromOutsideSphere) {
+
+  Sphere sphere(kVecZero, 1);
+  RayHitData hitData;
+  Ray alongYAxis(Vec3(0, -4, 0), Vec3(0, 1, 0));
+  ASSERT_TRUE(sphere.hitObject(alongYAxis, kSkipFreq, &hitData));
+  ASSERT_FLOAT_EQ(alongYAxis.origin().magnitude() - sphere.getRadius(),
+                  hitData.time);
+
+  Ray alongYAxisNoHit(Vec3(0, 4, 0), Vec3(0, 1, 0));
+  ASSERT_FALSE(sphere.hitObject(alongYAxisNoHit, kSkipFreq, &hitData))
+      << "hit at: " << hitData.collisionPoint();
+
+  Ray alongXAxis(Vec3(-4, 0, 0), Vec3(1, 0, 0));
+  ASSERT_TRUE(sphere.hitObject(alongXAxis, kSkipFreq, &hitData));
+  ASSERT_FLOAT_EQ(alongXAxis.origin().magnitude() - sphere.getRadius(),
+                  hitData.time);
+
+  Ray alongXAxisNoHit(Vec3(4, 0, 0), Vec3(1, 0, 0));
+  ASSERT_FALSE(sphere.hitObject(alongYAxisNoHit, kSkipFreq, &hitData))
+      << "hit at: " << hitData.collisionPoint();
+
+  Ray alongZAxis(Vec3(0, 0, -4), Vec3(0, 0, 1));
+  ASSERT_TRUE(sphere.hitObject(alongZAxis, kSkipFreq, &hitData));
+  ASSERT_FLOAT_EQ(alongZAxis.origin().magnitude() - sphere.getRadius(),
+                  hitData.time);
+
+  Ray alongZAxisNoHit(Vec3(0, 0, 4), Vec3(0, 0, 1));
+  ASSERT_FALSE(sphere.hitObject(alongZAxisNoHit, kSkipFreq, &hitData))
+      << "hit at: " << hitData.collisionPoint();
+
+  Vec3 randomChosenVec3(12.123, 5.623, 13.235);
+  Ray randomChosenOriginRay(randomChosenVec3,
+                            sphere.getOrigin() - randomChosenVec3);
+  ASSERT_TRUE(sphere.hitObject(randomChosenOriginRay, kSkipFreq, &hitData));
+  ASSERT_NEAR(randomChosenOriginRay.origin().magnitude() - sphere.getRadius(),
+              hitData.time, constants::kAccuracy);
+
+  Ray randomChosenOriginRayNoHit(randomChosenVec3,
+                                 randomChosenVec3 - sphere.getOrigin());
+  ASSERT_FALSE(
+      sphere.hitObject(randomChosenOriginRayNoHit, kSkipFreq, &hitData))
+      << "hit at: " << hitData.collisionPoint();
+
+  // This comes from proportions of the pitagorus triangle where sides satisfy
```

Comment:
What makes this test useful?

Reply:
its  not uefull in this case. This is the same case as first one ;) deleted it

Reply:
I still see it.

Reply:
?

Reply:
fixed. I deleted another test before, which was useless too :-P